contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,362 @@
+------------------------------------------------------------------------------
+You have specified this options:
+ $arguments
+------------------------------------------------------------------------------
+
+EOF
+  if [[ -n "$noop" ]]; then
+    echo "$0: --> Option -n was specified. I won't do anything."
+  fi
+fi
+
+################################
+# Check if tarball was specified
+################################
+export tarball="$1"
+
+if [[ ! -r "$tarball" ]]; then
+  echo "$0: FATAL: Can't find $tarball" 1>&2
+  exit 2
+fi
+verbose "Found tarball: $tarball"
+
+##############################
+# Get tarball name and version
+##############################
+export tardirname=$(basename "$tarball" | awk '{print substr ($0, 0, index($0, ".tar")-1)}')
+if [[ -z "$tardirname" ]]; then
+  echo "$0: FATAL: Cannot determine name from $tarball" 1>&2
+  exit 3
+fi
+verbose "Name of package: $tarball"
+
+export version=$(basename "$tardirname" | cut -d- -f2)
+if [[ -z "$tardirname" ]]; then
+  echo "$0: FATAL: Cannot determine version from $tarball" 1>&2
+  exit 4
+fi
+verbose "Version: $version"
+
+if [[ -n "$install_in_opt" ]]; then
+  prefix="/opt/libfabric/$version"
+  rpmbuild_options="$rpmbuild_options --define '_prefix $prefix'"
+  verbose "Setting RPM install path to: $prefix"
+fi
+
+######################################
+# Try to unpack spec file from tarball
+######################################
+if [[ -f "$specfile" ]]; then
+  if [[ -n "$verbose" ]]; then
+    echo "$0: WARNING: $specfile already exists and will be overwritten" 1>&2
+  fi
+fi
+
+if [[ -n "$unpack_spec" ]]; then
+  if [[ -z "$verbose" ]]; then
+    if [[ -z "$noop" ]]; then
+      tar -xf "$tarball"
+      cp -fp "$tardirname/$specfile" .
+      rm -rf "$tardirname"
+    fi
+  else
+    verbose "Extracting tarball"
+    runcmd "tar -xvf \"$tarball\""
+    verbose "Copying specfile"
+    runcmd "cp -fpv \"$tardirname/$specfile\" ."
+    verbose "Cleanup after extraction"
+    runcmd "rm -rfv \"$tardirname\""
+  fi
+fi
+
+##############################
+# Check if specfile is present
+##############################
+if [[ ! -r $specfile ]]; then
+  echo "$0: FATAL: Cannot find $specfile" 1>&2
+  if [[ -z "$noop" ]]; then
+    exit 5
+  fi
+else
+  if [[ -n "$verbose" ]]; then
+    echo "$0: --> Found specfile: $specfile"
+  fi
+fi
+
+#############################
+# Prepare directory for build
+#############################
+if [[ -z "$verbose" ]]; then
+  if [[ -z "$noop" ]]; then
+    rm -rf "$rpmbuilddir"
+    mkdir -p "$rpmbuilddir/SOURCES/"
+    cp -fp "$tarball" "$rpmbuilddir/SOURCES/"
+  fi
+else
+  verbose "$0: --> Cleanup old rpmbuild directory"
+  runcmd "rm -rfv \"$rpmbuilddir\""
+  verbose "$0: --> Create rpmbuild SOURCES directory"
+  runcmd "mkdir -pv \"$rpmbuilddir/SOURCES/\""
+  verbose "$0: --> Copy tarball to rpmbuild directory"
+  runcmd "cp -fpv \"$tarball\" \"$rpmbuilddir/SOURCES/\""
+fi
```

Comment:
I have adjusted this block a bit. I don't want to output messages when verbose is not specified, so I need 2 blocks.